Update bitflags to 0.4.0

signedin opened the pull request:
The bitflags version update is self explanatory; the new version 0.4.0 exists.

The gfx crate depends on draw_state so to test these changes the following actions were performed on gfx:
cargo build
cargo run --example triangle
cargo run --example cube
cargo run --example deferred
All of the above built successfully and the examples ran successfully.

signedin committed:
Update bitflags to 0.4.0

The bitflags version update is self explanatory; the new version 0.4.0 exists.

The gfx crate depends on draw_state so to test these changes the following actions were performed on gfx:
cargo build
cargo run --example triangle
cargo run --example cube
cargo run --example deferred
All of the above built successfully and the examples ran successfully.

bvssvni commented:
When updating the first non-zero number of any dependency it is a good idea to bump the first non-zero number in the version of the library, to avoid breaking existing code.

signedin committed:
Update Cargo.toml

Changed the first non-zero number of the library instead of the most minor number:
0.4.0 instead of 0.3.2.

kvark commented:
Is it really a breaking change though? I understand that bumping the major version every time is a safe bet, but in this case - what is exactly going to break?

bvssvni commented:
It breaks when Cargo tries to link two different versions of the same crate.

signedin commented:
I'd like to apologize for any confusion I caused.
I'm new to much of this environment.

In hindsight I should not have included any change to draw_state's package version. An appropriate person will handle that when they feel draw_state is ready to release a new crate. 

Also, while I tested against gfx, I failed to notice that other crates might depend on draw_state too.
I don't use piston but I see now via [https://crates.io/crates/draw_state/reverse_dependencies](https://crates.io/crates/draw_state/reverse_dependencies) that piston2d-graphics also depends on draw_state.

bvssvni commented:
@signedin You're right. piston2d-graphics was the library I was worried about. I opened https://github.com/gfx-rs/draw_state/issues/37 so we don't forget it. No need to apologize.
